cli/integration_tests/composable_config/composing-add-keys.t

```diff
@@ -0,0 +1,100 @@
+  apps/add-keys/out/
+  apps/add-keys/out/.keep
+  apps/add-keys/out/foo.min.txt
+
+# 2. Second run, test there was a cache hit (`cache` config`) and `output` was suppressed (`outputMode`)
+  $ ${TURBO} run add-keys-task --filter=add-keys
+  \xe2\x80\xa2 Packages in scope: add-keys (esc)
+  \xe2\x80\xa2 Running add-keys-task in 1 packages (esc)
+  \xe2\x80\xa2 Remote caching disabled (esc)
+  add-keys:add-keys-underlying-task: cache hit, replaying output 11b8af0d5b4ec46f
+  add-keys:add-keys-underlying-task: 
+  add-keys:add-keys-underlying-task: > add-keys-underlying-task
+  add-keys:add-keys-underlying-task: > echo "running add-keys-underlying-task"
+  add-keys:add-keys-underlying-task: 
+  add-keys:add-keys-underlying-task: running add-keys-underlying-task
+  add-keys:add-keys-task: cache hit, suppressing output d0092c043ad245ce
+  
+   Tasks:    2 successful, 2 total
+  Cached:    2 cached, 2 total
+    Time:\s*[\.0-9]+m?s >>> FULL TURBO (re)
+  
+# 3. Change input file and assert cache miss
+  $ echo "more text" >> $TARGET_DIR/apps/add-keys/src/foo.txt
+  $ ${TURBO} run add-keys-task --filter=add-keys
+  \xe2\x80\xa2 Packages in scope: add-keys (esc)
+  \xe2\x80\xa2 Running add-keys-task in 1 packages (esc)
+  \xe2\x80\xa2 Remote caching disabled (esc)
+  add-keys:add-keys-underlying-task: cache miss, executing b7390d43fae5a180
+  add-keys:add-keys-underlying-task: 
+  add-keys:add-keys-underlying-task: > add-keys-underlying-task
+  add-keys:add-keys-underlying-task: > echo "running add-keys-underlying-task"
+  add-keys:add-keys-underlying-task: 
+  add-keys:add-keys-underlying-task: running add-keys-underlying-task
+  add-keys:add-keys-task: cache miss, executing b256f31e3957fee3
+  add-keys:add-keys-task: 
+  add-keys:add-keys-task: > add-keys-task
+  add-keys:add-keys-task: > echo "running add-keys-task" > out/foo.min.txt
+  add-keys:add-keys-task: 
+  
+   Tasks:    2 successful, 2 total
+  Cached:    0 cached, 2 total
+    Time:\s*[\.0-9]+m?s  (re)
+  
+# 4. Set env var and assert cache miss
+  $ SOME_VAR=somevalue ${TURBO} run add-keys-task --filter=add-keys
+  \xe2\x80\xa2 Packages in scope: add-keys (esc)
+  \xe2\x80\xa2 Running add-keys-task in 1 packages (esc)
+  \xe2\x80\xa2 Remote caching disabled (esc)
+  add-keys:add-keys-underlying-task: cache hit, replaying output b7390d43fae5a180
+  add-keys:add-keys-underlying-task: 
+  add-keys:add-keys-underlying-task: > add-keys-underlying-task
+  add-keys:add-keys-underlying-task: > echo "running add-keys-underlying-task"
+  add-keys:add-keys-underlying-task: 
+  add-keys:add-keys-underlying-task: running add-keys-underlying-task
+  add-keys:add-keys-task: cache miss, executing 8e76bee49319e00e
+  add-keys:add-keys-task: 
+  add-keys:add-keys-task: > add-keys-task
+  add-keys:add-keys-task: > echo "running add-keys-task" > out/foo.min.txt
+  add-keys:add-keys-task: 
+  
+   Tasks:    2 successful, 2 total
+  Cached:    1 cached, 2 total
+    Time:\s*[\.0-9]+m?s  (re)
+  
```

Comment:
What if we used `dry-run` to reduce visual complexity for non-output-related tests? We're testing the hashes here, not the log messages.

```suggestion
# 3. Change input file and assert cache miss
  $ echo "more text" >> $TARGET_DIR/apps/add-keys/src/foo.txt
  $ ${TURBO} run add-keys-task --filter=add-keys --dry=json | jq -r ".tasks[].hash"
  b7390d43fae5a180
  b256f31e3957fee3

# 4. Set env var and assert cache miss
  $ SOME_VAR=somevalue ${TURBO} run add-keys-task --filter=add-keys --dry=json | jq -r ".tasks[].hash"
  b7390d43fae5a180
  8e76bee49319e00e
```